File: src/main/java/dev/rvbsm/fsit/command/PoseCommand.java
package dev.rvbsm.fsit.command;

import com.mojang.brigadier.Command;
import com.mojang.brigadier.context.CommandContext;
import dev.rvbsm.fsit.entity.PoseHandler;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;

import java.util.function.Consumer;

public record PoseCommand(String name, Consumer<PoseHandler> poseConsumer) implements Commandish<ServerCommandSource> {

	@Override
	public boolean requires(ServerCommandSource src) {
		return src.isExecutedByPlayer();
	}

	@Override
	public int executes(CommandContext<ServerCommandSource> ctx) {
		final ServerCommandSource src = ctx.getSource();
		final ServerPlayerEntity player = src.getPlayer();
		final PoseHandler poseHandler = (PoseHandler) player;
		if (player == null) return -1;

		if (poseHandler.isPosing()) poseHandler.resetPose();
		else poseConsumer.accept(poseHandler);

		return Command.SINGLE_SUCCESS;
	}
}

File: src/main/kotlin/dev/rvbsm/fsit/command/PoseCommand.kt
package dev.rvbsm.fsit.command

import com.mojang.brigadier.context.CommandContext
import dev.rvbsm.fsit.entity.PlayerPose
import dev.rvbsm.fsit.network.isInPose
import dev.rvbsm.fsit.network.resetPose
import dev.rvbsm.fsit.network.setPose
import net.minecraft.server.command.ServerCommandSource

enum class PoseCommand(private val pose: PlayerPose) : ModCommand<ServerCommandSource> {
    Sit(PlayerPose.Sitting),
    Crawl(PlayerPose.Crawling),
//    Lay(PlayerPose.Laying),
//    Spin(PlayerPose.Spinning)
    ;

    override fun requires(src: ServerCommandSource) = src.isExecutedByPlayer

    override fun executes(ctx: CommandContext<ServerCommandSource>): Int {
        val player = ctx.source.player ?: return -1
        if (player.hasVehicle()) return -1

        if (player.isInPose()) {
            player.resetPose()
        } else {
            player.setPose(pose)
        }

        return super.executes(ctx)
    }
}
